pkg/collector/process.go

```diff
@@ -145,22 +144,32 @@ func (cp *collectingProcess) decodeTemplateSet(templateBuffer *bytes.Buffer, obs
 		var elementLength uint16
 		err = util.Decode(templateBuffer, &elementid, &elementLength)
 		if err != nil {
-			return nil, fmt.Errorf("Error in decoding message: %v", err)
+			return nil, err
 		}
-		isIANARegistry := elementid[0] >> 7
-		if isIANARegistry != 1 {
+		isNonIANARegistry := elementid[0]>>7 == 1
+		if !isNonIANARegistry {
 			elementID = binary.BigEndian.Uint16(elementid)
-			enterpriseID = uint32(0)
+			enterpriseID = registry.IANAEnterpriseID
 			element, err = registry.GetInfoElementFromID(elementID, enterpriseID)
 			if err != nil {
 				return nil, err
 			}
 		} else {
+			/*
+					Encoding format for Enterprise-Specific Information Elements:
+				     0                   1                   2                   3
+					 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
```

Comment:
Format looks little off. You could view the file from github options "View File" at top right corner in the window.